src/pages/recipes/index.md

```diff
@@ -451,3 +451,394 @@ function main(){
+ 
+    //![remove event listener.]
+    app.removeEventListener("afterNew", myDisplayEventType);
+    //![remove event listener.]
+ 
+    myResolve();
+}
+ 
+function alert(msg)
+{
+    theDialog = app.dialogs.add();
+    col = theDialog.dialogColumns.add();
+    colText = col.staticTexts.add();
+    colText.staticLabel = "" + msg;
+    theDialog.canCancel = false;
+    theDialog.show();
+    theDialog.destroy();
+    return;
+}
+```
+
+## Menus
+InDesign scripting can add or remove menu items, perform any menu command, and attach scripts to menu items. This section shows how to work with InDesign menu scripting.
+
+Every menuItem is connected to a menuAction through the associatedMenuAction property. The properties of the menuAction define what happens when the menu item is selected. In addition to the menuActions defined by the user interface, users can create their scriptMenuActions, which associate a script with a menu selection.
+
+Here is a sample code for adding a menu item and associate an action to it. 
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+ 
+await main();
+async function main()
+{
+    return new Promise((resolve, reject) => {
+       mySnippet();
+  })
+}
+ 
+function mySnippet()
+{
+    //Adds a menu at the end of the main menu bar.
+    var mySampleScriptAction = app.scriptMenuActions.add("Display Message");
+ 
+    var myEventListener = mySampleScriptAction.eventListeners.add("onInvoke", function(){
+        alert("This menu item was added by a script.");
+    });
+ 
+    //If the submenu "Script Menu Action" does not already exist, create it.
+    try
+    {
+        var mySampleScriptMenu = app.menus.item("Main").submenus.item("Script Menu Action");
+    }
+    catch (myError)
+    {
+        var mySampleScriptMenu = app.menus.item("Main").submenus.add("Script Menu Action");
+    }
+ 
+    var mySampleScriptMenuItem = mySampleScriptMenu.menuItems.add(mySampleScriptAction);
+}
+ 
+function alert(msg)
+{
+    theDialog = app.dialogs.add();
+    col = theDialog.dialogColumns.add();
+    colText = col.staticTexts.add();
+    colText.staticLabel = "" + msg;
+    theDialog.canCancel = false;
+    theDialog.show();
+    theDialog.destroy();
+    return;
+}
+```
+
+## UXP InDesign DOM
+InDesign DOM provides APIs to create and modify InDesign application content through UXP scripting. You can find more details about this at [InDesign API](https://developer.adobe.com/indesign/dom/api/). 
+>
+Prior to 18.4, the InDesign DOM was available in the global space of the UXP script by default.
+>
+Starting 18.4, InDesign DOM is now available as a Javascript module, and it can be retrieved on a need basis using require() API. 
+
+### Overview
+The following code retrieves the InDesign DOM from 18.4 release onwards. This object allows you to open documents, modify them, run menu items, and more using the latest DOM version supported by the application.
+
+```js
+let id = require("indesign");
```

Comment:
This looks to be a breaking change. How do you plan to surface such radical changes to the developers?
At the least, please introduce this concept under Guides -> Platform Concepts. And find another place (such as Changelog) to announce such big changes.

Reply:
Please revisit all code snippets to ensure the `app` variable is declared and defined first.

Reply:
We can define the `app` variable only after defining the `id` variable. Will make sure all scripts have the `app` variable defined.

Reply:
Agreed

Reply:
@padmkris123, where can I find the Changelog?

Reply:
We need to create a new section for that. I think for now, under the References tab -> What's new should have the info. I recommend we use [this format](https://keepachangelog.com/en/1.0.0/). It categorically calls out the update, fixes, and changes.